MalwareAnalyzer/views.py

```diff
@@ -31,96 +31,68 @@ def apkid_analysis(app_dir, apk_file):
-
-        if "compiler" in apkid_dict["result"]:
-            apkid_dict["compiler"] = apkid_dict["result"]["compiler"]
-        else:
-            apkid_dict["compiler"] = ""
-
-        if "packer" in apkid_dict["result"]:
-            apkid_dict["packer"] = apkid_dict["result"]["packer"]
-        else:
-            apkid_dict["packer"] = ""
-
-        if "obfuscator" in apkid_dict["result"]:
-            apkid_dict["obfuscator"] = apkid_dict["result"]["obfuscator"]
-        else:
-            apkid_dict["obfuscator"] = ""
-
-        if "abnormal" in apkid_dict["result"]:
-            apkid_dict["abnormal"] = apkid_dict["result"]["abnormal"]
-        else:
-            apkid_dict["abnormal"] = ""
-
-        if "anti_disassembly" in apkid_dict["result"]:
-            apkid_dict["anti_disassembly"] = apkid_dict[
-                "result"]["anti_disassembly"]
-        else:
-            apkid_dict["anti_disassembly"] = ""
+        apkid_dict['anti_vm'] = apkid_dict['result'].get('anti_vm', '')    
+        apkid_dict['compiler'] = apkid_dict['result'].get('compiler', '')
+        apkid_dict['packer'] = apkid_dict['result'].get('packer', '')
+        apkid_dict['obfuscator'] = apkid_dict['result'].get('obfuscator', '')
+        apkid_dict['abnormal'] = apkid_dict['result'].get('abnormal', '')
+        apkid_dict['anti_disassembly'] = apkid_dict[
+                "result"].get('anti_disassembly', '')
+        apkid_dict['anti_debug'] = apkid_dict['result'].get('anti_debug', '')
+        apkid_dict['dropper'] = apkid_dict['result'].get('dropper','')
+        apkid_dict['manipulator'] = apkid_dict['result'].get('manipulator', '')
+        apkid_dict['result'] = ''
         
-        if "anti_debug" in apkid_dict["result"]:
-            apkid_dict["anti_debug"] = apkid_dict[
-                "result"]["anti_debug"]
-        else:
-            apkid_dict["anti_debug"] = ""
-
-        if "dropper" in apkid_dict["result"]:
-            apkid_dict["dropper"] = apkid_dict["result"]["dropper"]
-        else:
-            apkid_dict["dropper"] = ""
-
-        if "manipulator" in apkid_dict["result"]:
-            apkid_dict["manipulator"] = apkid_dict["result"]["manipulator"]
-        else:
-            apkid_dict["manipulator"] = ""
-
-        apkid_dict["result"] = ""
         if "files" in result:
-            apkid_dict["result"] = result["files"][1]["results"]
-        
-        if "anti_vm" in apkid_dict["result"]:
-            apkid_dict["anti_vm"] = apkid_dict["result"]["anti_vm"]
-        
-        if "compiler" in apkid_dict["result"]:
-            apkid_dict["compiler"] = apkid_dict["result"]["compiler"]
+            apkid_dict2["result"] = result["files"][1]["results"]
         
-        if "packer" in apkid_dict["result"]:
-            apkid_dict["packer"] = apkid_dict["result"]["packer"]
+        apkid_dict2['anti_vm'] = apkid_dict2['result'].get('anti_vm')  
+        if apkid_dict2['anti_vm'] is not None:

```

Comment:
@superpoussin22 How about`if 'anti_vm' in apkid_dict2`?

Reply:
@ajinabraham anti_vm should be in apkid_dict but I do a test just in case

Reply:
I soon as I know what is in which dictionnary (there's no spec) I will clean the code

Reply:
I have apdated it to make it more simpe